darts/models/forecasting/forecasting_model.py

```diff
@@ -871,13 +865,17 @@ def historical_forecasts(
                     if future_covariates_
                     else None,
                 ):
-                    self._fit_wrapper(
+                    # avoid fitting the same model multiple times
+                    model = self if not self._fit_called else self.untrained_model()
```

Comment:
I'm thinking we should always call `untrained_model()` even for the first pass here, WDYT?